### PR TITLE
operators [N] [CI] dell-csm-operator (0.1.0)

### PR DESCRIPTION
**Pull request by rensyct**
Signed-off-by: Rensy Thomas <rensy.thomas@dell.com>

Thanks submitting your Operator. Please check below list before you create your Pull Request.

### New Submissions

* [x] Are you familiar with our [contribution guidelines](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-via-pr.md)?
* [x] Have you [packaged and deployed](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md) your Operator for Operator Framework?
* [x] Have you tested your Operator with all Custom Resource Definitions?
* [x] Have you tested your Operator in all supported [installation modes](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#operator-metadata)?
* [x] Have you considered whether you want use [semantic versioning order](https://github.com/operator-framework/community-operators/blob/master/docs/operator-ci-yaml.md#semver-mode)?
* [x] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-prerequisites.md#sign-your-work)?
* [x] Is operator [icon](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#operator-icon) set?

### Updates to existing Operators

* [ ] Did you create a `ci.yaml` file according to the [update instructions](https://github.com/operator-framework/community-operators/blob/master/docs/operator-ci-yaml.md)?
* [ ] Is your new CSV pointing to the previous version with the `replaces` property if you chose `replaces-mode` via the `updateGraph` property in `ci.yaml`?
* [ ] Is your new CSV referenced in the [appropriate channel](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#channels) defined in the `package.yaml` or `annotations.yaml` ?
* [ ] Have you tested an update to your Operator when deployed via OLM?
* [ ] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-prerequisites.md#sign-your-work)?

### Your submission should not

* [x] Modify more than one operator
* [x] Modify an Operator you don't own
* [x] Rename an operator - please remove and add with a different name instead
* [x] Modify any files outside the above mentioned folders
* [x] Contain more than one commit. **Please squash your commits.**

### Operator Description must contain (in order)

1. [x] Description about the managed Application and where to find more information
2. [x] Features and capabilities of your Operator and how to use it
3. [x] Any manual steps about potential pre-requisites for using your Operator

### Operator Metadata should contain

* [x] Human readable name and 1-liner description about your Operator
* [x] Valid [category name](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#categories)<sup>1</sup>
* [x] One of the pre-defined [capability levels](https://github.com/operator-framework/operator-courier/blob/4d1a25d2c8d52f7de6297ec18d8afd6521236aa2/operatorcourier/validate.py#L556)<sup>2</sup>
* [x] Links to the maintainer, source code and documentation
* [x] Example templates for all Custom Resource Definitions intended to be used
* [x] A quadratic logo

Remember that you can preview your CSV [here](https://operatorhub.io/preview).

--

<sup>1</sup> If you feel your Operator does not fit any of the pre-defined categories, file an issue against this repo and explain your need

<sup>2</sup> For more information see [here](https://sdk.operatorframework.io/docs/overview/#operator-capability-level)

### CODE REVIEW

**Comment by rensyct** [comment]
Hi @J0zi , please let me know if there is something missing from our end that is causing the tests to fail

**Comment by J0zi** [comment]
@rensyct Directory ''/tmp/community-operators-for-catalog/operators/dell-csm-operator/0.1.0/tests/scorecard/'' specified in ''operators.operatorframework.io.test.config.v1'' doesn''t exists

**Comment by rensyct** [comment]
Hi @mvalarh, all prow tests are failing, Summary in logs show that the operator is up and running. Please let know if there is something missing in operator as the error in logs do not point out any error in operators

File: operators/dell-csm-operator/0.1.0/manifests/dell-csm-operator.v0.1.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,769 @@
...
+  - Powerscale
+  - Authorization
+  links:
+  - name: Documentation
+    url: https://dell.github.io/csm-docs/docs/deployment/csmoperator/
+  maintainers:
+  - email: dell.csm.operator@dell.com
+    name: Dell CSM Operator
+  maturity: alpha
+  minKubeVersion: 1.21.0
```
Comment (J0zi) [inline operators/dell-csm-operator/0.1.0/manifests/dell-csm-operator.v0.1.0.clusterserviceversion.yaml:765]: @rensyct `minKubeVersion: 1.21.0` should mean releasing to `OCP v4.8` and higher. Please add `annotations.com.redhat.openshift.versions: "v4.8"` to `annotations.yaml`

Comment (rensyct) [reply operators/dell-csm-operator/0.1.0/manifests/dell-csm-operator.v0.1.0.clusterserviceversion.yaml:765]: Hi @J0zi, support matrix for dell-csm-operator is OCP 4.8 and OCP 4.9. Added `com.redhat.openshift.versions: v4.8  ` to `annotations.yaml`

Comment (rensyct) [reply operators/dell-csm-operator/0.1.0/manifests/dell-csm-operator.v0.1.0.clusterserviceversion.yaml:765]: Hi @J0zi 
We use the `replaces` field in CSV. Will the upgrade from `CSM 0.1.0 to CSM 0.2.0 `work as expected on` OCP 4.10` with `replaces` field in CSV if we change it to `4.8-4.9`

Comment (J0zi) [reply operators/dell-csm-operator/0.1.0/manifests/dell-csm-operator.v0.1.0.clusterserviceversion.yaml:766]: @rensyct  If I understand it correctly, CSM works also on `v4.10`, so we can merge it with `annotations.com.redhat.openshift.versions: v4.8`, which will provide your operator with all indexes `4.8+`. Please confirm, that I can merge it as it is.

Comment (rensyct) [reply operators/dell-csm-operator/0.1.0/manifests/dell-csm-operator.v0.1.0.clusterserviceversion.yaml:766]: Please merge.. thank you @J0zi

Comment (J0zi) [reply operators/dell-csm-operator/0.1.0/manifests/dell-csm-operator.v0.1.0.clusterserviceversion.yaml:766]: @rensyct your `olm.maxOpenShiftVersion` is `4.9`, so the label should be aligned with this value. I suggest `annotations.com.redhat.openshift.versions: 4.8-4.9` which will not release to `v4.10` index. 

Or you should change `  olm.properties: '[{"type": "olm.maxOpenShiftVersion", "value": "4.9"}]'` if you want release in `v4.10` index

Comment (J0zi) [reply operators/dell-csm-operator/0.1.0/manifests/dell-csm-operator.v0.1.0.clusterserviceversion.yaml:766]: let me summarize:
```
minKubeVersion: 1.21.0
annotations.com.redhat.openshift.versions: 4.8-4.9
'[{"type": "olm.maxOpenShiftVersion", "value": "4.9"}]'
```
this is all aligned ^ and means 4.8-4.9

this is not compatible:
```
minKubeVersion: 1.21.0
annotations.com.redhat.openshift.versions: 4.8
'[{"type": "olm.maxOpenShiftVersion", "value": "4.9"}]'
```

if you are not using deprecated API, you can omit the line containing [{"type": "olm.maxOpenShiftVersion", "value": "4.9"}]


so this is also compatible and means 4.8+:
```
minKubeVersion: 1.21.0
annotations.com.redhat.openshift.versions: 4.8
```

Comment (rensyct) [reply operators/dell-csm-operator/0.1.0/manifests/dell-csm-operator.v0.1.0.clusterserviceversion.yaml:766]: For this we dont have 4.10 in the support matrix. so it is right that it should not release to 4.10 index. But next release can we use "replaces" keyword in CSV. Would the operator work as expected on a OCP .10

Comment (rensyct) [reply operators/dell-csm-operator/0.1.0/manifests/dell-csm-operator.v0.1.0.clusterserviceversion.yaml:766]: We are not using deprecated APIs. i can omit the below line
[{"type": "olm.maxOpenShiftVersion", "value": "4.9"}]

**Comment by rensyct** [comment]
Thank you @J0zi for merging the PR

**Comment by J0zi** [comment]
Please check again

**Comment by rensyct** [comment]
The operator started listing from yesterday. Thank you @J0zi

**Pull request merged**